### PR TITLE
3.0 path resolution improvements

### PR DESCRIPTION
**Pull request by srbaker**

### CODE REVIEW

**Comment by thobe** [comment]
that's a lot of tests being removed. scary.

**Comment by srbaker** [comment]
@thobe It is. They're the ones that were added, for the hack whichis removed here. :)

**Comment by benbc** [comment]
@srbaker It would be helpful, I think, to explain why in the commit message for 8b85300.

**Comment by benbc** [comment]
@srbaker I think you can remove the second argument to `resolve_path` in 8b85300, can't you?

**Comment by benbc** [comment]
@srbaker Two comments above. Otherwise this looks fine.

File: community/server/src/main/java/org/neo4j/server/ServerCommandLineArgs.java [hunk 1]
```diff
@@ -43,6 +43,7 @@
 public class ServerCommandLineArgs
```
Comment (thobe) [inline community/server/src/main/java/org/neo4j/server/ServerCommandLineArgs.java:43]: Do you think you could fix the javadoc here while you're at it?
- `--config` is never used.
- `-C` is never used
- `--config-dir` is not documented
- `--home-dir` is not documented

**Comment by benbc** [comment]
This can be simplified: https://github.com/srbaker/neo4j/blob/2ba7388785a19f56aff556eca3fae92925682e50/packaging/standalone/src/main/distribution/shell-scripts/bin/neo4j#L219. Just `CONSOLE_LOG="${NEO4J_LOGS}/neo4j.log"`.

File: packaging/standalone/src/main/distribution/shell-scripts/bin/neo4j-shared.sh [hunk 2]
```diff
@@ -126,8 +124,8 @@ _setup_calculated_paths() {
   if [[ -z "${NEO4J_HOME:-}" ]]; then
     NEO4J_HOME="$(cd "$(dirname "$0")"/.. && pwd)"
   fi
-  : "${NEO4J_CONF:=conf}"
-  readonly NEO4J_ROOT NEO4J_CONF
+  : "${NEO4J_CONF:=$(resolve_path conf)}"
```
Comment (benbc) [inline packaging/standalone/src/main/distribution/shell-scripts/bin/neo4j-shared.sh]: There is no need for `resolve_path` here: `: "${NEO4J_CONF:="${NEO4J_HOME}/conf"}"`.

File: packaging/standalone/src/main/distribution/shell-scripts/bin/Neo4j-Management/Get-Java.ps1 [hunk 3]
```diff
@@ -202,7 +202,7 @@ Function Get-Java
           $ShellArgs += "-XX:NumberOfGCLogFiles=5"
         }
       }
-      $ShellArgs += @("-Dfile.encoding=UTF-8",$serverMainClass,"--config-dir=$($Neo4jServer.ConfDir)")
+      $ShellArgs += @("-Dfile.encoding=UTF-8",$serverMainClass,"--config-dir=$($Neo4jServer.ConfDir)","--home-dir=$($Neo4jServer.Home)")
```
Comment (benbc) [inline packaging/standalone/src/main/distribution/shell-scripts/bin/Neo4j-Management/Get-Java.ps1:205]: No test for this?

**Pull request merged**